[MRG+1] Return list instead of 3d array for MultiOutputClassifier.predict_proba

pjbull opened the pull request:
#### Reference Issue
Fixes #8093 

#### What does this implement/fix? Explain your changes.
Returns a list of the predicted probabilities for each estimator in the`MultiOutputClassifier` rather than trying to stack them into a 3d matrix.

#### Any other comments?
This is the same behavior now as the `RandomForestClassifier`, which handles data of this shape without being wrapped in the `MultiOutputClassifier`.

pjbull committed:
Return list instead of 3d array for MultiOutputClassifier.predict_proba

jnothman commented on sklearn/multioutput.py:
```diff
@@ -214,16 +214,18 @@ def predict_proba(self, X):
         -------
-        T : (sparse) array-like, shape = (n_samples, n_classes, n_outputs)
-            The class probabilities of the samples for each of the outputs
+        p : array of shape = [n_samples, n_classes], or a list of n_outputs
```
for this type spec to render correctly as one line, you need a `\` at the end of the line.

jnothman commented on sklearn/tests/test_multioutput.py:
```diff
@@ -150,6 +153,39 @@ def test_multiclass_multioutput_estimator():
 
+def test_multiclass_multioutput_estimator_predict_proba():
+    # make test deterministic
+    rs = np.random.RandomState(123456)
```
By convention we use `rng`, not `rs`

jnothman reviewed (changes requested):
Otherwise LGTM, thanks!

pjbull committed:
Update flake8, docstring, variable name

 - Changed `rs` to `rng` to follow convention.
 - Made sure changes were flake8 approved
 - Add `\` to continue docstring for `predict_proba` return value.

pjbull commented:
New commit has the requested changes.

pjbull committed:
Sub random.choice for np.random.choice

`np.random.choice` isn’t available in Numpy 1.6, so opt for the Python
version instead.

pjbull committed:
Make test labels deterministic

pjbull commented:
CI passing, should be good to go 🏇

TomDLT commented on test_output.log:
```diff
@@ -0,0 +1,1535 @@
+rm -f tags
```
you probably need to remove this file

TomDLT reviewed (changes requested):
Remove sneaky log file and it's good to me

jnothman reviewed (approved):
LGTM

jnothman commented:
Of course, after you remove the .log.

Please add an entry to what's new. Or even one under API changes and one under bug fixes.

pjbull committed:
Remove hanging chad...

pjbull committed:
Add bug fix and API change to whats new

raghavrv commented:
Thanks @pjbull... Merging once appveyor becomes green...

raghavrv committed:
Merge branch 'master' into multiclass-predict-proba

jnothman commented:
Thanks for identifying and fixing this, @pbull
